### DISCUSSION TITLE
What exactly does AttributeNumeric ?

### DISCUSSION THREAD

**Discussion by BenOnTrack** [systemed/tilemaker#733] [Q&A]
I'm trying to store as an INT a parameter (based on the OSM ID) and realized that:

```
function WriteOSMidAndType(id_prefix)
	local id_str = id_prefix .. Id()
	local id_int = tonumber(id_str)
	AttributeNumeric("id", id_int)
	Attribute("id2", id_str)
end
```

"id" and "id2" are not the same...
I'm interested into "id" as I'd like to handle number for ID rather than string.

Is there anything linked to the "size" of the string being converted into a number ? 

When I print both parameters, id_str == id_int
I also confirm that "id2" is correct, while "id" is not - potentially due to a conversion made by AttributeNumeric function

I'm confused :)

**Comment by systemed** [comment]
It's just to do with how the value is encoded in the vector tile. Using `Attribute` stores it as a string (i.e. the characters "1","2" and "3") whereas `AttributeNumeric` stores it efficiently as a series of bytes.

If you use a tool such as vt2geojson to decompile the vector tile, you should find that anything written with `Attribute` is represented in JSON as (say) `"id": "123"`, whereas if written with `AttributeNumeric` it would be `"id": 123`.

**Reply by BenOnTrack** [reply]
I understand, but it does not work for me.

If you check the 1st post, I'm trying to append a "prefix" (1,2 or3) to a number -> so I initially convert everything as a String, and then re-convert into an int. 

When storing that string with `Attribute`  I have "id":"2539990" (which is correct), and when storing with `AttributeNumeric` i have "id":2541004

**Reply by systemed** [reply]
I'm not aware of any issues with AttributeNumeric changing the value (and I use it in my own profile without issues) but that doesn't mean there aren't any. I'd suggest producing a very simple testcase that eliminates the possibility of an issue in the Lua code.
